Tests.Integration.Presentation.Web/Tools/Internal/UI_Configuration/UIConfigurationHelper.cs

```diff
@@ -3,37 +3,84 @@
         {
-            var cookie = optionalLogin ?? await HttpApi.GetCookieAsync(OrganizationRole.GlobalAdmin);
-            var uri = TestEnvironment.CreateUrl($"api/v1/organizations/{organizationId}/ui-config/modules/{module}");
+            var cookie = optionalLogin ?? await HttpApi.GetCookieAsync(OrganizationRole.LocalAdmin);
             return await HttpApi.PutWithCookieAsync(TestEnvironment.CreateUrl($"api/v1/organizations/{organizationId}/ui-config/modules/{module}"), cookie, body);
         }
 
-        public static async Task<UIModuleCustomizationDTO> GetAllAsync(int organizationId, string module, Cookie optionalLogin = null)
+        public static async Task<UIModuleCustomizationDTO> GetCustomizationByModuleAsync(int organizationId, string module)
         {
-            using var response = await SendGetRequestAsync(organizationId, module, optionalLogin);
+            using var response = await SendGetRequestAsync(organizationId, module);
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
             return await response.ReadResponseBodyAsKitosApiResponseAsync<UIModuleCustomizationDTO>();
         }
 
+        public static async Task<UIModuleCustomizationDTO> CreateUIModuleAndSaveAsync(int organizationId, string module,
+            Cookie optionalLogin = null)
+        {
+            var dto = PrepareTestUiModuleCustomizationDto();
+            using var response = await SendPutRequestAsync(organizationId, module, dto, optionalLogin);
+
+            Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
+            return dto;
+        }
+
         public static async Task<HttpResponseMessage> SendGetRequestAsync(int organizationId, string module, Cookie optionalLogin = null)
         {
-            var cookie = optionalLogin ?? await HttpApi.GetCookieAsync(OrganizationRole.GlobalAdmin);
+            var cookie = optionalLogin ?? await HttpApi.GetCookieAsync(OrganizationRole.LocalAdmin);
 
             return await HttpApi.GetWithCookieAsync(TestEnvironment.CreateUrl($"api/v1/organizations/{organizationId}/ui-config/modules/{module}"), cookie);
         }
+        
+        public static UIModuleCustomizationDTO PrepareTestUiModuleCustomizationDto(int numberOfElements = 1, string key = "", bool isEnabled = false)
+        {
+            return new UIModuleCustomizationDTO
+            {
+                Nodes = PrepareTestNodes(numberOfElements, key, isEnabled)
+            };
+        }
+        
+        public static List<CustomizedUINodeDTO> PrepareTestNodes(int numberOfElements = 1, string key = "", bool isEnabled = false, params string[] keys)
+        {
+            var nodes = new List<CustomizedUINodeDTO>();
+            for (var i = 1; i <= numberOfElements; i++)
+            {
+                key = string.IsNullOrEmpty(key) ? RandomString(i) : key;
+                nodes.Add(new CustomizedUINodeDTO { Key = key, Enabled = isEnabled });
+                key = "";
```

Comment:
what is the purpose of the "key" parameter. It seems like it will reset it (changing a parameter state is always bad practice) on every iteration. If what you mean is "prefix" then use it as such and then apply a suffix on each iteration making it unique